include/xtensor/xstrided_view.hpp

```diff
@@ -153,10 +153,36 @@ namespace xt
 
         using inner_storage_type = typename base_type::inner_storage_type;
         using storage_type = typename base_type::storage_type;
-        using linear_iterator = typename storage_type::iterator;
-        using const_linear_iterator = typename storage_type::const_iterator;
-        using reverse_linear_iterator = std::reverse_iterator<linear_iterator>;
-        using const_reverse_linear_iterator = std::reverse_iterator<const_linear_iterator>;
+
+        template <class C, class = void_t<>>
+        struct get_linear_iterator : std::false_type
+        {
+            using iterator = typename C::iterator;
+        };
+
+        template <typename C>
+        struct get_linear_iterator<C, void_t<decltype(std::declval<C>().linear_begin())>> : std::true_type
+        {
+            using iterator = typename C::linear_iterator;
+        };
+
+        template <class C, class = void_t<>>
+        struct get_const_linear_iterator : std::false_type
+        {
+            using iterator = typename C::const_iterator;
+        };
+
+        template <typename C>
+        struct get_const_linear_iterator<C, void_t<decltype(std::declval<C>().linear_cbegin())>> : std::true_type
+        {
+            using iterator = typename C::const_linear_iterator;
+        };
```

Comment:
These structures should be defined outside of the class (maybe in the detail namespace for now).